File: Modules/SEDO/EMS/SIE.Web.EMS.Purchases/FixtureAcceptances/Commands/SaveAcceptanceAttachmentCommand.cs
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.EMS.Purchases.FixtureAcceptances;
using SIE.Web.Command;
using System;

namespace SIE.Web.EMS.Purchases.FixtureAcceptances.Commands
{
    /// <summary>
    /// 保存命令
    /// </summary>
    public class SaveAcceptanceAttachmentCommand : FormSaveCommand
    {
        /// <summary>
        /// 保存前动作
        /// </summary>
        protected override void OnSaving(Entity entity)
        {
            base.OnSaving(entity);
        }
        /// <summary>
        /// 执行上传和复传
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var data = args.Data.ToJsonObject<FixtureAcceptanceAttachment>();
            data.PersistenceStatus = PersistenceStatus.New;

            RF.Save(data);

            return true;
        }

    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS.Purchases/FixtureAcceptances/Commands/SaveAcceptanceAttachmentCommand.js
SIE.defineCommand('SIE.Web.EMS.Purchases.FixtureAcceptances.Commands.SaveAcceptanceAttachmentCommand', {
    extend: 'SIE.cmd.FormSave',
    meta: { text: "上传", group: "edit", iconCls: "icon-SaveEntity icon-blue" },
    canExecute: function (view) {        
        return this.callParent(arguments);
    }, 
    execute: function (view, source) {
        var data = view.getCurrent().data;
        
        view.execute({
            withIds: true,
            data: data,            
            success: function (rst) {
                var win = Ext.getCmp('UploadFixtureAcceptancesCommand_Window');
                win.close();
                CRT.Event.fire("SIE.EMS.Purchases.FixtureAcceptances.FixtureAcceptance_refresh");
                Ext.MessageBox.alert("提示".t(), "上传成功".t());
            }
        });
    }
});